packages/core/register/src/hook.js

```diff
@@ -0,0 +1,115 @@
+import Module from 'module';
+import process from 'process';
+import path from 'path';
+import fs from 'fs';
+import {addHook} from 'pirates';
+
+import Parcel, {Asset, createDependency} from '@parcel/core';
+import Cache from '@parcel/cache';
+import syncPromise from '@parcel/utils/lib/syncPromise';
+
+let hooks = {};
+const originalRequire = Module.prototype.require;
+const DEFAULT_CLI_OPTS = {
+  watch: false
+};
+
+// Cache this to make it a lil faster?
+function readPkgUp(dir) {
+  let dirContent = fs.readdirSync(dir);
+
+  for (let filename of dirContent) {
+    if (filename === 'package.json') {
+      return JSON.parse(fs.readFileSync(path.join(dir, 'package.json')));
+    }
+  }
+
+  if (dir === '/') {
+    return null;
+  }
+
+  return readPkgUp(path.dirname(dir));
+}
+
+function isParcelDep(filename) {
+  let pkg = readPkgUp(path.dirname(filename));
+  return pkg && (pkg.name.includes('@parcel') || pkg.name.includes('parcel'));
```

Comment:
I assume the second check is for third party plugins? I guess I'd look for `pkg.name.startsWith('@parcel') || pkg.name.startsWith('parcel-')` to be more safe.